src/decoder/types.rs

```diff
@@ -46,133 +59,102 @@ pub struct Parameters {
     pub traits_base: Vec<TraitSchema>,
 }
 
-#[cfg_attr(test, derive(serde::Serialize, Clone))]
-#[derive(serde::Deserialize, PartialEq, Eq)]
-pub enum ArgsType {
-    String,
-    Number,
-}
-
 #[cfg_attr(test, derive(serde::Serialize, Clone))]
 #[derive(serde::Deserialize)]
 pub enum Pattern {
     Options,
     Range,
-    Raw,
+    RawNumber,
+    RawString,
     Utf8,
 }
 
 #[cfg_attr(test, derive(serde::Serialize, Clone))]
 #[derive(serde::Deserialize)]
 pub struct TraitSchema {
     pub name: String,
-    pub type_: ArgsType,
+    pub type_: String,
     pub offset: u64,
     pub len: u64,
     pub pattern: Pattern,
-    pub args: Option<Vec<String>>,
+    pub args: Option<Vec<Value>>,
 }
 
 #[cfg(test)]
 impl TraitSchema {
     #[allow(dead_code)]
     pub fn new(
-        name: &str,
-        type_: ArgsType,
+        name: String,
+        type_: String,
         offset: u64,
         len: u64,
         pattern: Pattern,
-        args: Option<Vec<&str>>,
+        args: Option<Vec<Value>>,
     ) -> Self {
         Self {
-            name: name.to_owned(),
+            name,
             type_,
             offset,
             len,
             pattern,
-            args: args.map(|v| v.into_iter().map(ToOwned::to_owned).collect()),
+            args,
         }
     }
 
     #[allow(dead_code)]
-    pub fn encode(&self) -> Vec<Value> {
+    pub fn encode(self) -> Vec<Value> {
         let mut values = vec![
-            Value::String(self.name.clone()),
-            Value::String(match self.type_ {
-                ArgsType::String => "string".to_owned(),
-                ArgsType::Number => "number".to_owned(),
-            }),
+            Value::String(self.name),
+            Value::String(self.type_),
             Value::Number(self.offset.into()),
             Value::Number(self.len.into()),
             Value::String(match self.pattern {
                 Pattern::Options => "options".to_owned(),
                 Pattern::Range => "range".to_owned(),
-                Pattern::Raw => "raw".to_owned(),
+                Pattern::RawNumber => "rawNumber".to_owned(),
+                Pattern::RawString => "rawString".to_owned(),
```

Comment:
changes in here could impact our exported DOB/0 protocol format, if ok, I'm fine, besides, what about this name format: `raw/number`?

Reply:
I think it's fine since this won't impact those existed decoders.
`rawNumber` is one byte shorter~